### ISSUE TITLE
Log files with version inside

### ISSUE THREAD

**Issue by cdamken** [owncloud/ios-app#444]
People send me Log files and there is no line where it says which version is e testing.

It will be AMAZING if you add one line in the log files that says the version.

@michaelstingl @hosy

**Comment by michaelstingl** [comment]
Cool idea. This is it how it looks like in the desktop log:

```
$ /Applications/owncloud.app/Contents/MacOS/owncloud --logfile - --logdebug
m_crashReporterChar:  /Applications/owncloud.app/Contents/MacOS/owncloud_crash_reporter
m_crashReporterWChar: /Applications/owncloud.app/Contents/MacOS/owncloud_crash_reporter
07-08 15:38:32:245 [ info gui.application ]:	"################## ownCloud locale:[en_DE] ui_lang:[] version:[2.6.0alpha1 (build 11615)] os:[macOS Mojave (10.14)]"
07-08 15:38:32:245 [ info gui.application ]:	Using "en_DE" translation
07-08 15:38:32:246 [ info gui.application ]:	Adding extra plugin search path: "/Applications/owncloud.app/Contents/MacOS/../PlugIns"
07-08 15:38:32:247 [ info gui.application ]:	VFS suffix plugin is available
07-08 15:38:32:290 [ info gui.socketapi ]:	server started, listening at  "4AP2STM4H5.com.owncloud.desktopclient.socketApi"
07-08 15:38:32:290 [ info gui.folder.manager ]:	setting remote poll timer interval to 30000 msec
07-08 15:38:32:900 [ debug gui.activity ]	[ OCC::ActivitySettings::setNotificationRefreshInterval ]:	Starting Notification refresh timer with  300  sec interval
07-08 15:38:32:901 [ info gui.application ]:	Tray menu workarounds: noabouttoshow: true fakedoubleclick: false showhide: false manualvisibility: true
07-08 15:38:32:914 [ info gui.folder.manager ]:	Setup folders from settings file
07-08 15:38:32:915 [ info plugins ]:	Created VFS instance from plugin "owncloudsync_vfs_suffix"
07-08 15:38:32:915 [ debug gui.folder ]	[ OCC::Folder::checkLocalPath ]:	Checked local path ok
07-08 15:38:32:915 [ info sync.configfile ]:	Adding system ignore list to csync: "/Applications/owncloud.app/Contents/Resources/sync-exclude.lst"
07-08 15:38:32:915 [ info sync.configfile ]:	Adding user defined ignore list to csync: "/Users/mstingl_oc/Library/Preferences/ownCloud/sync-exclude.lst"
```

You also see the server version:
```
07-08 15:38:34:114 [ info sync.networkjob.checkserver ]:	status.php returns:  QJsonDocument({"edition":"Community","installed":true,"maintenance":false,"needsDbUpgrade":false,"productname":"ownCloud","version":"10.2.1.4","versionstring":"10.2.1"})   QNetworkReply::NoError  Reply:  QNetworkReplyHttpImpl(0x7fec1c56c740)
07-08 15:38:34:114 [ info sync.connectionvalidator ]:	** Application: ownCloud found:  QUrl("https://owncloud.mst.works")  with version  "10.2.1" ( "10.2.1.4-ownCloud" )
07-08 15:38:34:114 [ info sync.connectionvalidator ]:	QUrl("https://owncloud.mst.works") has server version "10.2.1.4-ownCloud"
07-08 15:38:34:115 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job OCC::CheckServerJob finished for "status.php"
```

**Comment by michaelstingl** [comment]
Show the same info as in the setting screen? (including the Git hash)

**Comment by hosy** [comment]
@cdamken the version info for the iOS app oder for the server?

**Comment by michaelstingl** [comment]
> @cdamken the version info for the iOS app oder for the server?

Primarily the app version, maybe the iOS version. ( `… version:[2.6.0alpha1 (build 11615)] os:[macOS Mojave (10.14)]"` )

Server version would be nice, but I'd assume it's already somewhere in the logs…

**Comment by felix-schwarz** [comment]
@cdamken @michaelstingl Implemented this in #446. It logs a line like this near the top of every log file:

`2019-07-08 23:08:38.667000+0200 ownCloud[1253:538893] [info] | [LogIntro] Host: com.owncloud.ios-app 1.0.2 (125) #ea6c365; SDK: 1.0 (1) #49664d1; OS: iOS 12.3.1; Device: iPhone (iPhone8,1); Localizations: [de-DE, de]; Log options: level=Debug, destinations=["writer.stderr", "writer.file"], options=["option.log-requests-and-responses"], maskPrivateData=false [OCLogFileWriter.m:103|FULL]`

Since the line is tagged with `LogIntro`, it can be found in log files with a simple search.

**Comment by jesmrec** [comment]
Done in #446

**Comment by michaelstingl** [comment]
@felix-schwarz Please backport to the 1.0.4 release branch

**Comment by michaelstingl** [comment]
released with 1.0.4
